app/commands/donations/payment_intent/handle_invoice_failure.rb

```diff
@@ -1,38 +1,39 @@
-module Donations
-  module PaymentIntent
-    class HandleInvoiceFailure
-      include Mandate
-
-      def initialize(id: nil, invoice: nil)
-        raise "Specify either id or invoice" unless id || invoice
-
-        @id = id
-        @invoice = invoice
-      end
-
-      # We guard against spammers here
-      def call
-        return unless user
-        return unless user.stripe_customer_id.present?
-        return if user.uid # Return if the user has auth'd via GitHub
-
-        has_three_failed_invoiced_today = Stripe::Charge.search(
-          query: %(customer:"#{user.stripe_customer_id}" AND status:"failed" AND created>#{(Time.current - 24.hours).to_i}),
-          limit: 3
-        ).count == 3
-
-        user.update!(disabled_at: Time.current) if has_three_failed_invoiced_today
-      end
-
-      private
-      attr_reader :id, :invoice
-
-      memoize
-      def user
-        raise "No customer in the invoice" unless invoice.customer
-
-        User.find_by(stripe_customer_id: invoice.customer)
-      end
-    end
+class Donations::PaymentIntent::HandleInvoiceFailure
+  include Mandate
+
+  initialize_with id: nil, invoice: nil do
+    raise "Specify either id or invoice" unless id || invoice
+  end
+
+  # We guard against spammers here
+  def call
+    return unless user
+    return if user.uid # Return if the user has auth'd via GitHub
+
+    user.update!(disabled_at: Time.current) if too_many_failed_invoices_in_last_24_hours?
+  end
+
+  private
+  attr_reader :id, :invoice
+
+  def too_many_failed_invoices_in_last_24_hours?
+    number_of_failed_invoices_in_last_24_hours >= MINIMUM_FAILED_INVOICES_IN_LAST_24_HOURS
   end
+
+  def number_of_failed_invoices_in_last_24_hours
+    Stripe::Charge.search(
+      query: %(customer:"#{user.stripe_customer_id}" AND status:"failed" AND created>#{(Time.current - 24.hours).to_i}),
+      limit: 3
```

Comment:
```suggestion
      limit: MINIMUM_FAILED_INVOICES_IN_LAST_24_HOURS
```

Reply:
If we increased `MINIMUM_FAILED_INVOICES_IN_LAST_24_HOURS` to 4, then the logic would break as this only retrieves 3

Reply:
Lol